### PR TITLE
Add option to suppress creation of displaced mesh

### LINKED ISSUE #12580: Option to prevent creation of displaced mesh
## Rationale

We need a way to force the displaced mesh to not be created in some cases, even when `displacements` is set.

## Description

In mechanics problems, we use `displacements` in so many places that we almost always set it as a global parameter. However, in small strain calculations, we don't need the displaced mesh to be created, and it can cause side effects. The most important of these is that we may invert elements in the displaced mesh, which wouldn't be an issue if it didn't exist in the first place.

We can accomplish this by adding a `use_displaced_mesh` option to the Mesh block. It would default to `true`, which would give the current behavior. If it were set to false, a displaced mesh would not be created, even if the user provided a `displacements` parameter in that block.

## Impact

This would just add an additional optional parameter to the Mesh block. No existing behavior should be changed.

### LINKED ISSUE #12584: Expand elastic patch tests to include other options
## Rationale

The tensor_mechanics module has a set of elastic patch test problems that all use finite strain. They should be expanded to include a complete set of modeling assumptions including small total strain and small incremental strain.

## Description

See above.

## Impact

No impact on users. Just add more test options.

### PR DESCRIPTION
**Pull request by bwspenc**
closes #12584
closes #12580

### CODE REVIEW

File: framework/src/actions/SetupMeshAction.C [hunk 1]
```diff
@@ -208,14 +215,14 @@ SetupMeshAction::act()
...
-    if (isParamValid("displacements"))
+    if (isParamValid("displacements") && (getParam<bool>("use_displaced_mesh") == true))
       _displaced_mesh = _factory.create<MooseMesh>(_type, "displaced_mesh", _moose_object_pars);
   }
   else if (_current_task == "init_mesh")
   {
     _mesh->init();
 
-    if (isParamValid("displacements"))
+    if (isParamValid("displacements") && (getParam<bool>("use_displaced_mesh") == true))
```
Comment (dschwen) [inline framework/src/actions/SetupMeshAction.C]: Did you want to change this to
```
if (isParamValid("displacements") && getParam<bool>("use_displaced_mesh"))
```
?

Comment (bwspenc) [reply framework/src/actions/SetupMeshAction.C]: Yes -- just getting to it right this second, coincidentally.

Comment (bwspenc) [reply framework/src/actions/SetupMeshAction.C]: Ok, I just changed those.

**Review by dschwen — APPROVED** [review]
Looks good to me

**Pull request merged**